tests/ui/associated-types/issue-54182-1.rs

```diff
@@ -21,12 +21,12 @@ macro_rules! overload {
 }
 
 fn main() {
-    let () = overload!(42, true);
+    let _: () = overload!(42, true);
```

Comment:
i dont think this is an improvement, the lint should not lint here

Reply:
(This is from earlier version of the lint, I forgor to change it)